process-lib/src/process.ml

```diff
@@ -633,22 +637,22 @@ let run_exit_status =
            Printf.ksprintf failwith "%s: command not found" (quote_for_errors prog))
   in
   fun prog args -> pack2 prim prog args
-
-let run_exit_code prog args =
-  run_exit_status prog args >>| function
+    
+let command_exit_code cmd_string prog args =
+  command_exit_status cmd_string prog args >>| function
```

Comment:
this needs to be:

```ocaml
let command_exit_code cmd_string =
  let cmd = command_exit_status cmd_string in
  fun prog args -> ...
```

to avoid calling `Prim.make` over and over. `Prim.make` should actually return a `('a, 'b) t Staged.

But given that it's the same primitive for each of the three functions, it feels like this should be a functor taking a `sig val cmd : string end` argument and returning the three functions.